src/posthog-surveys.ts

```diff
@@ -27,6 +28,23 @@ export class PostHogSurveys {
         }
     }
 
+    getMatchingUrl(url?: string): boolean {
```

Comment:
I believe this will actually require a version update for users in order for it to work for them? @neilkakkar

Reply:
Yes

Reply:
I'd recommend making this helper pure, i.e pass in URL and regex to match. Makes it easier to test, and remove reference to `window.location.href` from this function

Reply:
Sounds good! Does it make sense for that helper to be in `utils.ts` rather than a class method? At that point it would be a simple regex test of an argument url and pattern.

Reply:
Yep good call